packages/manager/apps/pci-object-storage/src/api/hooks/useUser.ts

```diff
@@ -0,0 +1,67 @@
+import { useQueries, useQuery } from '@tanstack/react-query';
+import { ColumnSort, PaginationState } from '@ovh-ux/manager-react-components';
+import { applyFilters, Filter } from '@ovh-ux/manager-core-api';
+import { useMemo } from 'react';
+import { getAllUsers, getS3Credentials, TUser } from '@/api/data/user';
+import { paginateResults, sortResults } from '@/helpers';
+
+export const useAllUsers = (projectId: string) =>
+  useQuery({
+    queryKey: ['project', projectId, 'users'],
+    queryFn: () => getAllUsers(projectId),
+  });
+
+export const useUsers = (projectId: string) => {
+  const { data: users, isPending } = useAllUsers(projectId);
+  return useQueries({
+    queries: (users || [])?.map((user) => ({
+      queryKey: ['user', user.id, 's3Credentials'],
```

Comment:
Maybe adding the projectId to the queryKey, since the fetching concerne a specific project